### PR TITLE
feat: default signer on bee instance

### PR DESCRIPTION
**Pull request by AuHau**
Introduce default signer for Bee instance.

### CODE REVIEW

File: test/bee-class.spec.ts [hunk 1]
```diff
@@ -350,4 +350,38 @@ describe('Bee class', () => {
...
+  describe('signer', () => {
+    const socHash = '9d453ebb73b2fedaaf44ceddcf7a0aa37f3e3d6453fea5841c31f0ea6d61dc85' as HexString
+
+    it('should be possible to pass it in constructor', async () => {
+      const identifier = makeBytes(32) // all zeroes
+      const socAddress = makeSOCAddress(identifier, makeEthAddress(testIdentity.address))
+      await tryDeleteChunkFromLocalStorage(socAddress)
+
+      const bee = new Bee(BEE_URL, { signer: testIdentity.privateKey })
+      const socWriter = bee.makeSOCWriter()
```
Comment (agazso) [inline test/bee-class.spec.ts:363]: It would be good to have similar tests for `makeFeedWriter`

**Review by agazso — APPROVED** [review]
Looks good, I would add tests for the feed writers as well.

File: test/bee-class.spec.ts (hunk 1 above)
Comment (AuHau) [reply test/bee-class.spec.ts:363]: Hmm not sure what value would that bring. I am sort of "abusing" the `socWriter` here to verify that the correct signer is used with it. The main tested logic is directly in the `Bee` class and is shared between both SOC and Feed logic (the `resolveSigner()` method).

Comment (agazso) [reply test/bee-class.spec.ts:363]: Ideally with the changes in this PR these are the potential cases to be tested when calling `makeSOCWriter`:
- no signer in constructor, no signer provided as argument -> error
- signer in constructor, no signer provided as argument -> signer in constructor called
- singer in constructor, signer provided as argument -> signer from argument called

And similarly these cases tested with `makeFeedWriter`.

Comment (AuHau) [reply test/bee-class.spec.ts:363]: Well, exactly these test cases are already present. I don't think it make much sense to test it also with `makeFeedWriter` but it is not problem for me, so I will add these.

File: src/types/index.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { BeeError } from '../utils/error'
 import type { AxiosRequestConfig } from 'axios'
 import { HexString } from '../utils/hex'
+import { Signer } from '../chunk/signer'
```
Comment (nugaon) [inline src/types/index.ts:4]:
```suggestion
import type { Signer } from '../chunk/signer'
```

File: src/bee.ts [hunk 3]
```diff
@@ -38,26 +39,34 @@ import { assertReference } from './utils/type'
...
- *
- * @param url URL of a running Bee node
  */
 export class Bee {
   public readonly url: string
+  public readonly signer?: Signer
 
-  constructor(url: string) {
+  /**
+   * @param url URL of a running Bee node
```
Comment (nugaon) [inline src/bee.ts:48]: This is the Bee client's API URL, please correct this jsdoc.
In an other PR we need to make clear distinction between `apiUrl` and `debugUrl` in codes which face to devs.

File: src/bee.ts [hunk 4]
```diff
@@ -38,26 +39,34 @@ import { assertReference } from './utils/type'
...
+   * @param options
+   * @param options.signer Optional Signer object/string/Uint8Array that will be default signer for the instance.
```
Comment (nugaon) [inline src/bee.ts:50]:
```suggestion
   * @param [options.signer] Signer object or private key of the Signer in form of either hex string or Uint8Array that will be default signer for the instance.
```

Please describe what exactly the object/string/Uint8Array in this context.

File: src/bee.ts [hunk 5]
```diff
@@ -38,26 +39,34 @@ import { assertReference } from './utils/type'
...
- * @param url URL of a running Bee node
  */
 export class Bee {
   public readonly url: string
+  public readonly signer?: Signer
 
-  constructor(url: string) {
+  /**
+   * @param url URL of a running Bee node
+   * @param options
```
Comment (nugaon) [inline src/bee.ts:49]:
```suggestion
   * @param [options]
```

**Review by nugaon — APPROVED** [review]
I left some comments, please consider those, but seems good to me.

File: src/bee.ts (hunk 5 above)
Comment (AuHau) [reply src/bee.ts:49]: What is the functionality of the square brackets?

**Pull request merged**